harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -334,8 +335,10 @@ def _compute_validation_metrics(self) -> workload.Response:
             batch_metrics = []
 
             self.validation_loader = cast(torch.utils.data.DataLoader, self.validation_loader)
-            check.gt(len(self.validation_loader), 0)
-            for batch in self.validation_loader:
+            val_loader_len = len(self.validation_loader)
+            check.gt(val_loader_len, 0)
+            batch_stop_idx = val_loader_len if not self.context.env.test_mode else 1
+            for batch in islice(self.validation_loader, 0, batch_stop_idx):
```

Comment:
non-blocking:  What if you just put an `if self.env.test_mode: break` at the end of the loop?  I had never heard of `islice()` before, and while it is definitely not wrong to use advanced features of a language, in this case the simple thing seems so simple it's at least worth suggesting.

Reply:
originally I was designing it to allow arbitrary batch length but early breaking is simpler 👍🏼